tests/components/anova_sous_vide/test_init.py

```diff
@@ -0,0 +1,47 @@
+"""Test init for Anova Sous Vide."""
+
+from unittest.mock import patch
+
+from anova_wifi import AnovaOffline
+
+from homeassistant.components.anova_sous_vide import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import STATE_UNAVAILABLE
+from homeassistant.core import HomeAssistant
+
+from . import async_init_integration, create_entry
+
+
+async def test_async_setup_entry(hass: HomeAssistant):
+    """Test a successful setup entry."""
+    await async_init_integration(hass)
+    state = hass.states.get("sensor.mode")
+    assert state is not None
+    assert state.state != STATE_UNAVAILABLE
+    assert state.state == "Low water"
+
+
+async def test_config_not_ready(hass: HomeAssistant):
+    """Test for setup failure if connection to Anova is missing."""
+    entry = create_entry(hass)
+    with patch(
+        "anova_wifi.AnovaPrecisionCooker.update",
+        side_effect=AnovaOffline(),
+    ):
+        entry.add_to_hass(hass)
+        await hass.config_entries.async_setup(entry.entry_id)
+        assert entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_unload_entry(hass: HomeAssistant):
+    """Test successful unload of entry."""
+    entry = await async_init_integration(hass)
+
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert entry.state is ConfigEntryState.LOADED
+
+    assert await hass.config_entries.async_unload(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert entry.state is ConfigEntryState.NOT_LOADED
+    assert not hass.data.get(DOMAIN)
```

Comment:
Please do not interact with `hass.data` in tests

https://developers.home-assistant.io/docs/development_testing/?_highlight=writing&_highlight=tes#writing-tests-for-integrations

Reply:
I removed line 47, I did a quick search around and it seems like most integrations either use the outdated (not hass.data.get(DOMAIN)) or just don't have it. If there is a third option that would be better practice, I would be happy to correct it.

Reply:
Removing it is just fine. 👍

Reply:
This is partially unrelated to this PR, but would it be appropriate for me to create a PR where I remove all of the assert not hass.data.get(DOMAIN) in test_unload_entry 's?

I know myself personally when I am looking to add a test, I look at other implementations. So if this is no longer valid, I could remove it.

But if you think it A) Isn't worth it or B) Not my place to do something that like, then I will not.